File: server/schemas/relatorio.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TrilhaBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    nome: str


class RelatorioCreate(BaseModel):
    periodo_inicio: datetime
    periodo_fim: datetime


class RelatorioItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    trilha_id: int
    aceites: int
    rejeicoes: int
    trilha: TrilhaBrief


class RelatorioResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    gerado_por_id: int
    periodo_inicio: datetime
    periodo_fim: datetime
    total_solicitacoes: int
    total_aceites: int
    total_rejeicoes: int
    created_at: datetime
    itens: list[RelatorioItemResponse]

File: front-end/src/app/models/relatorio.ts
export interface TrilhaBrief {
  id: number;
  nome: string;
}

export interface RelatorioItem {
  trilha_id: number;
  aceites: number;
  rejeicoes: number;
  trilha: TrilhaBrief;
}

export interface Relatorio {
  id: number;
  gerado_por_id: number;
  periodo_inicio: string;
  periodo_fim: string;
  total_solicitacoes: number;
  total_aceites: number;
  total_rejeicoes: number;
  created_at: string;
  itens: RelatorioItem[];
}

export interface RelatorioCreateRequest {
  periodo_inicio: string;
  periodo_fim: string;
}
